File: output/llama_4_maverick/avatar/algo_based_translation/java/python/failed_translation/codeforces_30_A.py

def main():
    A, B, n = map(int, input().split())
    if A == 0:
        if B == 0:
            print(1)
        else:
            print("No solution")
        return
    if B % A != 0:
        print("No solution")
        return
    B //= A
    neg = B < 0
    if neg and n % 2 == 0:
        print("No solution")
        return
    if neg:
        B = -B
    for x in range(int(B) + 1):
        if x ** n == B:
            if neg:
                print(-x)
            else:
                print(x)
            return
    print("No solution")

if __name__ == "__main__":
    main()

File: output/qwen_2.5_72b_instruct/avatar/algo_based_translation/python/java/failed_translation/codeforces_30_A.java

import java.util.Scanner;

public class codeforces_30_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String[] input = scanner.nextLine().split(" ");
        int A = Integer.parseInt(input[0]);
        int B = Integer.parseInt(input[1]);
        int n = Integer.parseInt(input[2]);
        System.out.println(task4(A, B, n));
    }

    public static Object task4(int A, int B, int n) {
        for (int X = -1000; X <= 1000; X++) {
            if (A * Math.pow(X, n) == B) {
                return X;
            }
        }
        return "No solution";
    }
}

